src/main/java/io/confluent/idesidecar/restapi/kafkarest/TopicV3ApiImpl.java

```diff
@@ -31,12 +46,30 @@ public Uni<Void> deleteKafkaTopic(String clusterId, String topicName) {
   public Uni<TopicData> getKafkaTopic(
       String clusterId, String topicName, Boolean includeAuthorizedOperations
   ) {
-    return topicManager.getKafkaTopic(clusterId, topicName, includeAuthorizedOperations);
+    return topicManager
+        .getKafkaTopic(clusterId, topicName, includeAuthorizedOperations)
+        .onItem()
+        .transform(topicDescription -> fromTopicDescription(clusterId, topicDescription));
   }
 
   @Override
   public Uni<TopicDataList> listKafkaTopics(String clusterId, Boolean includeAuthorizedOperations) {
-    return topicManager.listKafkaTopics(clusterId, includeAuthorizedOperations);
+    return topicManager.listKafkaTopics(clusterId, includeAuthorizedOperations)
+            .onItem().transformToUni(topicDescriptionMap -> uniItem(TopicDataList
+            .builder()
+            .kind("KafkaTopicList")
+            .metadata(ResourceCollectionMetadata
+                .builder()
+                .next(null)
+                .self(forTopics(clusterId).getRelated())
+                .build()
+            )
+            .data(topicDescriptionMap
+                .stream()
+                .map(t -> fromTopicDescription(clusterId, t))
+                .toList()
+            ).build())
+        );
```

Comment:
Might be easier to read:

```suggestion
    return topicManager
        .listKafkaTopics(clusterId, includeAuthorizedOperations)
        .onItem()
        .transformToUni(topicDescriptionMap ->
            uniItem(
                TopicDataList
                  .builder()
                  .kind("KafkaTopicList")
                  .metadata(
                      ResourceCollectionMetadata
                        .builder()
                        .next(null)
                        .self(forTopics(clusterId).getRelated())
                        .build()
                  )
                  .data(
                      topicDescriptionMap
                        .stream()
                        .map(topic -> fromTopicDescription(clusterId, topic))
                        .toList()
                  )
                  .build()
            )
        );
```